packages/api/amplify_api/lib/src/api_plugin_impl.dart

```diff
@@ -51,6 +53,10 @@ class AmplifyAPIDart extends AmplifyAPI {
   final AWSHttpClient? _baseHttpClient;
   late final AmplifyAuthProviderRepository _authProviderRepo;
 
+  /// A function which creates a stream of booleans representing network connectivity
+  /// at the hardware level.
+  Stream<bool> Function()? networkStreamGenerator;
```

Comment:
I would create an interface:

```dart
enum ConnectivityStatus {
  connected,
  disconnected,
}

abstract class ConnectivityInterface {
  Stream<ConnectivityResult> get onConnectivityChanged;
}
```

and use an enum instead of a boolean. A boolean locks us into two values and is not very descriptive.

You could even add a third `unknown` value, if that's helpful, and make that the default e.g.

```dart
enum ConnectivityStatus {
  unknown,
  connected,
  disconnected,
}

class ConnectivityInterface {
  const ConnectivityInterface();  

  Stream<ConnectivityResult> get onConnectivityChanged => const Stream.value(unknown);
}
```